### PR TITLE
refactor(browser-platform-dynamic): remove support for passing in pla…

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/platform-browser-dynamic/index.ts [hunk 1]
```diff
@@ -212,6 +212,7 @@ export function bootstrap<C>(
     precompile = normalizeArray(customProvidersOrDynamicModule.precompile);
     compiler = customProvidersOrDynamicModule.compiler;
   }
+  let deprecationMessages: string[] = [];
```
Comment (vicb) [inline modules/@angular/platform-browser-dynamic/index.ts]: `const`, same below in updated code

File: modules/@angular/platform-browser/test/browser/bootstrap_spec.ts [hunk 2]
```diff
@@ -77,10 +77,18 @@ class HelloUrlCmp {
   greeting = 'hello';
 }
 
-@Component({
-  selector: 'hello-app',
-  template: `<div  [title]="'HELLO' | lowercase"></div><div *ngIf="show"></div>`
-})
+@Directive({selector: '[someDir]', host: {'[title]': 'someDir'}})
```
Comment (vicb) [inline modules/@angular/platform-browser/test/browser/bootstrap_spec.ts]: should we factorize somewhere ?

File: modules/@angular/platform-browser-dynamic/index.ts [hunk 3]
```diff
@@ -223,20 +224,44 @@ export function bootstrap<C>(
...
+    let console = inj.get(Console, null);
+    if (console) {
+      compilerProviders.push([{provide: Console, useValue: console}]);
     }
   }
   if (!compiler) {
     compiler = browserCompiler({providers: compilerProviders});
   }
+  deprecationMessages.forEach((msg) => {
+    let console: Console = compiler.injector.get(Console);
```
Comment (vicb) [inline modules/@angular/platform-browser-dynamic/index.ts]: outside of the loop ?

Is there some more detailed doc we can link for the deprecation ? (commit comment, release note, ... ?)

**Comment by vicb** [comment]
LGTM, a few comments in the first commit

File: modules/@angular/platform-browser-dynamic/index.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/platform-browser-dynamic/index.ts]: Done

File: modules/@angular/platform-browser/test/browser/bootstrap_spec.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/platform-browser/test/browser/bootstrap_spec.ts]: Would make it harder to read / understand. This is shared between platform-browser and compiler-cli...

File: modules/@angular/platform-browser-dynamic/index.ts (hunk 3 above)
Comment (tbosch) [inline modules/@angular/platform-browser-dynamic/index.ts]: Done

**Pull request closed without merging**